src/unity/python/turicreate/data_structures/sframe.py

```diff
@@ -4992,17 +4993,24 @@ def unpack(self, column_name=None, column_name_prefix=None, column_types=None,
         """
         if column_name is None:
             if self.num_columns()==0:
-                raise RuntimeError("no column exists in the current SFrame")
-            if self.num_columns()==1:
-                column_name=self.column_names()[0]
-                if column_name_prefix==None:
+                raise RuntimeError("No column exists in the current SFrame")
+            unpack_columns,column_number=0,0
+            for t in range(self.num_columns()):
+                column_type = self.column_types()[t]
+                if column_type==type(dict()) or column_type==type(list()) or column_type==type(array.array('c','he')):
+                    
+                    if column_name is None:
+                        column_name = self.column_names()[t]
+                    else:
+                        raise RuntimeError("Column name needed to unpack")
+            if column_name is not None and column_name_prefix is None:
                     column_name_prefix=""
-            else:
-                raise RuntimeError("column name needed to unpack")
-        
+
+            if column_name is None:
+                raise RuntimeError("No columns can be unpacked")
 
         if column_name not in self.column_names():
```

Comment:
Nit: maybe this should go into an `elif` from the `if` statement in line 4994, since if we deduced the column name automatically, then it's definitely in `self.column_names()`. What do you think?